### ISSUE TITLE
[NPC/Vendor] I can't interact with the armor and repair vendor

### ISSUE THREAD

**Issue by Fixerage** [WoWManiaUK/Blackwing-Lair#5067]
**Links:**

from WoWHead or our Armory

**What is happening:**  

I have a DK lvl 62 and i did a quest with a dragon  evreytime the npcs kept hitting me until my death (just a one bit was enough to take me down) . I died 9 times in unless than an hour after that my gear has a red mark so i need to repair them I've searched for a repair vendor and i used right click to open it but nothing happens i reloged and that npc and other npcs wasn't appear i closed the game and reopen it they appeared but i can't right click the vendors to open anything

**What should happen:**
I should right click and the window appear so i can repair my gear
**Is it crashing the server?:**
No, 
**Other Information:**
The place is under the ship the second place i go to for questing

**Comment by Echinaceax** [comment]
Quest  https://wowhead.com/quest=12779
Video 
https://youtu.be/Aad4pi6VV4s?t=1381
https://www.wowhead.com/quest=12779/an-end-to-all-things#videos:id=54245

I seen myself,  this quest does not behave like in the videos, (ours is a little more agressive) thats for sure @sam-ozak maybe a few adustments needed, and the vender is being checked by a helpful ingame player later 


- [ ] Check vendor reporter mentions

edit
tester also on it :)

**Comment by Elvorfindra** [comment]
https://www.wowhead.com/npc=28500/master-siegesmith-corvus

I tested the starting area with troll dk up to the quest "An end to all things..." . Before this quest start NPCs reacting normal (you can repair). After turning completed quest in there will be only selling/buying NPC reacting to you, not anyone else (not even xp boost goblin). 

What the repair says:
![WoWScrnShot_021021_161234](https://user-images.githubusercontent.com/77320686/107535184-cd77b780-6bc0-11eb-8d15-f48cf8dee264.jpg)

Sell/buy: 
![WoWScrnShot_021021_161258](https://user-images.githubusercontent.com/77320686/107535297-e7b19580-6bc0-11eb-88f4-2e09dd2e9a8a.jpg)
![WoWScrnShot_021021_161307](https://user-images.githubusercontent.com/77320686/107535447-0748be00-6bc1-11eb-8e7b-b291e69f13b8.jpg)
![WoWScrnShot_021021_161312](https://user-images.githubusercontent.com/77320686/107535473-0ca60880-6bc1-11eb-844d-830101d9d9e8.jpg)
![WoWScrnShot_021021_161317](https://user-images.githubusercontent.com/77320686/107535498-13348000-6bc1-11eb-97e9-aa7e262a60e4.jpg)


This is the Npcs not responding/responding .

**Comment by Fixerage** [comment]
Not inside the ship under it the land under it

**Comment by Echinaceax** [comment]
MORE INFOMATION NEEDED @sam-ozak 

1. Vender name ???

**Comment by Elvorfindra** [comment]
https://www.wowhead.com/npc=28760/hargus-the-gimp
Hargus the Gimp is the vendor name at Deaths Breach. He’s not reacting aswell!

**Comment by Echinaceax** [comment]
https://wowwiki-archive.fandom.com/wiki/Hargus_the_Gimp

![image](https://user-images.githubusercontent.com/39439201/107567502-0d04ca80-6be6-11eb-9d0f-c62f45e91219.png)

**Comment by mfranca0009** [comment]
Is Hargus the Gimp the only vendor not reacting? @Echinaceax 

We need full list of vendors tested that are not reacting so I can fix all of them at once. From OP, it sounds like there is multiple vendors not working; I need the links (entry IDs) for all of them.

**Comment by Echinaceax** [comment]
all of the above @Elvorfindra said

**Comment by mfranca0009** [comment]
So far, we have two NPCs not functioning properly after a certain quest has been completed.

### On Ship
https://www.wowhead.com/npc=28500/master-siegesmith-corvus

### On Ground (Off Ship)
https://www.wowhead.com/npc=28760/hargus-the-gimp

Need to confirm these are the ONLY two or if there is more having problems.

**Comment by Elvorfindra** [comment]
That is correct @mfranca0009 ! Those 2 NPC are the ones that should do repairs but are not working!

**Comment by mfranca0009** [comment]
Okay thanks 👍  @Elvorfindra

**Comment by mfranca0009** [comment]
No problems using these vendors...they do work. I did change it to be more blizzlike, actually showing their gossip options instead of opening directly to their vendor store.

The only issue I could see is at quest, https://www.wowhead.com/quest=12779/an-end-to-all-things, and when turning it in, you're at a different phase then the default phase you start in for DKs. 

You end up in phase 175 where you only see the Lich King standing on his pedestal, while all other NPCs are gone. At this point, you shouldn't be able to see `Hargus The Gimp` because he's not in phase 175. If you were able to see him, this is probably why he wasn't reacting to you; somehow your phasing was buggy where you weren't even suppose to be seeing him but you did.

With `Master Siegesmith Corvus`, had no issues interacting with this vendor.

Is `Hargus The Gimp` suppose to be phased out while the Lich King is on his pedestal at the `An End To All Things` quest?

**Comment by Echinaceax** [comment]
@mfranca0009 thanks 
then we just send the tweaks you did then leave the rest
